kroxylicious-enforcer-rules/src/main/java/io/kroxylicious/enforcer/rules/SystemTestIsolationRule.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.enforcer.rules;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Named;
+
+import org.apache.maven.enforcer.rule.api.EnforcerRule;
+import org.apache.maven.enforcer.rule.api.EnforcerRuleException;
+import org.apache.maven.enforcer.rule.api.EnforcerRuleHelper;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.execution.ProjectDependencyGraph;
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.component.configurator.expression.ExpressionEvaluationException;
+
+/**
+ * System test isolation rule
+ * Checks that no other modules depends on System test module
+ */
+@Named("systemTestIsolationRule") // rule name - must start from lowercase character
+public class SystemTestIsolationRule implements EnforcerRule {
```

Comment:
I had just expected a ban on other modules importing `io.kroxylicious.systemtests` (possibly configured in parent pom) though this works. 

And does give us a spot for other custom rules which we might well want.

Reply:
I follow the idea of having an Enforcer Rule and I found it pretty easy to be removed from Kroxylicious when the system test will be moved to another project.

Anyway, I will investigate if this ban can be configured in the parent pom. I didn't find a better solution for that than the enforcer rule.

Reply:
I'm surprised/disappointed too that we need so much stuff to enforce a rule like this.   But if it is really the only way.

Reply:
Yeah, we use that in the [additional filters](https://github.com/kroxylicious/kroxylicious/blob/f3c3bbbcb56d6c9ed5569d1a9ddf4747d6acef20/kroxylicious-additional-filters/pom.xml#L32-L57 ) module to ensure we only use the public APIs there.

I read this rule as trying to do the reverse though, its trying to ensure that nothing in the reactor pulls this the system tests module but thinking again it does end up being the same...

Reply:
@franvila when your not travelling what are your thoughts about https://github.com/franvila/kproxy/pull/52 ?

@k-wall this [PR](https://github.com/franvila/kproxy/pull/52) is inline with what I initially expected, but I wonder if we should also expand it to cover things like the perftests and integration tests?